oas_docs/README.md

```diff
@@ -1,9 +1,164 @@
+| `api-docs-lint-errs`       | Lints built result bundles for errors                                                                                                                                                                              |
+| `api-docs-preview`         | Generates (ESS + Serverless) Kibana OpenAPI bundles preview                                                                                                                                                        |
+| `api-docs-overlay`         | Applies [overlays](https://docs.bump.sh/help/specification-support/overlays/) from `overlays` folder to the Kibana OpenAPI bundles and generate `*.new.yaml` files. Overlays help to fine tune the result bundles. |
+| `api-docs-overlay-preview` | Generates a preview for bundles produced by `api-docs-overlay`                                                                                                                                                     |
+
+## Bundling automation
+
+To have always up to date Kibana OpenAPI bundles bundling runs as a part of `Checks` step in CI on every PR build and on merge builds. If there are any changes in source OpenAPI files these changes get to the updated Kibana OpenAPI bundles and CI commits the changes. In that case CI is marked as failed and needs to be restarted.
+
+## Bundling solution domains
+
+### Security Solution
+
+Security Solution uses **specification first approach**. There are multiple OpenAPI files describing individual API endpoints. It helps to enable code generation besides automatic API reference documentation. That OpenAPI files are bundled into domain OpenAPI bundles by `kbn-openapi-bundler`. Paths to produced Security Solution domain OpenAPI bundles were added to the merge scripts `merge_ess_oas.js` and `merge_serverless_oas.js` to include these bundles in Kibana OpenAPI bundle. There are the following API domains in Security Solution included in the Kibana OpenAPI bundle
+
+#### Security AI Assistant
+
+Root folder `x-pack/packages/kbn-elastic-assistant-common`.
+
+|                             |                                                                                |
+| --------------------------- | ------------------------------------------------------------------------------ |
+| Source OpenAPI files folder | `impl`                                                                         |
+| Bundling script             | `scripts/openapi/bundle.js`                                                    |
+| Result Serverless bundle    | `docs/openapi/serverless/elastic_assistant_api_2023_10_31.bundled.schema.yaml` |
+| Result ESS bundle           | `docs/openapi/ess/elastic_assistant_api_2023_10_31.bundled.schema.yaml`        |
```

Comment:
There's a lot of tables here and perhaps too much details - some of this information can become outdated in the future. Let's try to make it a bit more succinct and resilient to refactorings. What do you think about a simple unordered list:

- Security AI Assistant API
  - Bundling script: `x-pack/packages/kbn-elastic-assistant-common/scripts/openapi/bundle.js`
  - Bundles' location: `x-pack/packages/kbn-elastic-assistant-common/docs/openapi/{ess|serverless}`
- Security Detections API
  - Bundling script: `x-pack/plugins/security_solution/scripts/openapi/bundle_detections.js`
  - Bundles' location: `x-pack/plugins/security_solution/docs/openapi/{ess|serverless}`
- etc